spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/pulsar/PulsarAutoConfiguration.java

```diff
@@ -131,7 +132,7 @@ PulsarTemplate<?> pulsarTemplate(PulsarProducerFactory<?> pulsarProducerFactory,
 
 	@Bean
 	@ConditionalOnMissingBean(PulsarConsumerFactory.class)
-	DefaultPulsarConsumerFactory<Object> pulsarConsumerFactory(PulsarClient pulsarClient,
+	DefaultPulsarConsumerFactory<?> pulsarConsumerFactory(PulsarClient pulsarClient,
```

Comment:
Might be a good idea to add a test (or adjust existing tests) for each of these scenarios in `org.springframework.boot.autoconfigure.pulsar.PulsarAutoConfigurationTests`.

There are tests that verify custom user beans are respected, but they inject in `PulsarConsumerFactory<String>` rather than attempt to use a specific type of these changed beans (DPCF, CPLCF). (e.g. `org.springframework.boot.autoconfigure.pulsar.PulsarAutoConfigurationTests.ConsumerFactoryTests#whenHasUserDefinedBeanDoesNotAutoConfigureBean`).